server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1449,16 +1461,15 @@ public static boolean isRepositoryInUse(ClusterState clusterState, String reposi
      * @param snapshot   snapshot
      * @param listener   listener
      * @param repositoryStateId the unique id representing the state of the repository at the time the deletion began
-     * @param version minimum ES version the repository should be readable by
+     * @param state      cluster state
      */
     private void deleteSnapshotFromRepository(Snapshot snapshot, @Nullable ActionListener<Void> listener, long repositoryStateId,
-                                              Version version) {
+                                              ClusterState state) {
```

Comment:
can we avoid passing the full cluster state?

The cluster state is possibly a very large object, and holding onto it for a longer time has caused issues in the past, given that we go onto a different thread here, with a potentially large queue of other snapshot-related tasks to execute first.

Reply:
Jup done, that was certainly a little too dry :)